File: android/app/src/main/java/com/claudewidget/data/UsageData.kt
package com.claudewidget.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.OffsetDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.Locale

@Serializable
data class UsagePeriod(
    val utilization: Double,
    @SerialName("resets_at") val resetsAt: String?
) {
    /** Progress bar fraction: 0.0 to 1.0 */
    val fraction: Double
        get() = (utilization / 100.0).coerceIn(0.0, 1.0)

    /** Integer percentage for display: 0 to 100 */
    val percent: Int
        get() = utilization.toInt().coerceIn(0, 100)

    /** "Mon 9:00 AM" in user's local timezone; "soon" if parsing fails or resets_at is null */
    fun formatResetTime(): String {
        val raw = resetsAt ?: return "soon"
        return try {
            val dt = OffsetDateTime.parse(raw)
            val formatter = DateTimeFormatter.ofPattern("EEE h:mm a", Locale.getDefault())
            dt.atZoneSameInstant(ZoneId.systemDefault()).format(formatter)
        } catch (_: Exception) {
            "soon"
        }
    }
}

@Serializable
data class UsageResponse(
    @SerialName("five_hour") val fiveHour: UsagePeriod,
    @SerialName("seven_day") val sevenDay: UsagePeriod
)

/**
 * Wraps the API response with a local fetch timestamp.
 * Not serialized to JSON -- stored as two fields in DataStore.
 */
data class UsageData(
    val response: UsageResponse,
    val fetchedAt: Long = System.currentTimeMillis()
)

File: ios/ClaudeUsage/Data/UsageData.swift
import Foundation

struct UsagePeriod: Codable {
    let utilization: Double
    let resetsAt: String

    /// Progress bar fraction: 0.0 to 1.0
    var fraction: Double {
        min(max(utilization / 100.0, 0.0), 1.0)
    }

    /// Integer percentage for display: 0 to 100
    var percent: Int {
        Int(min(max(utilization, 0.0), 100.0))
    }

    /// Parsed Date from resetsAt string; nil if parsing fails
    var resetDate: Date? {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        return formatter.date(from: resetsAt)
    }

    /// "Mon 1:00 PM" in the user's local timezone; "soon" if parsing fails
    var resetFormatted: String {
        guard let date = resetDate else { return "soon" }
        let formatter = DateFormatter()
        formatter.dateFormat = "EEE h:mm a"
        formatter.timeZone = .current
        return formatter.string(from: date)
    }
}

struct UsageResponse: Codable {
    let fiveHour: UsagePeriod
    let sevenDay: UsagePeriod
}

struct UsageData: Codable {
    let response: UsageResponse
    var fetchedAt: Date = .now
}
